src/hardware/ne2000.cpp

```diff
@@ -1689,8 +1519,8 @@ class NE2K: public Module_base {
 	}	
 	
 	~NE2K() {
-		if(adhandle) pcap_close(adhandle);
-		adhandle=0;
+		if(ethernet != 0) delete ethernet;
```

Comment:
Pointless `if` statement. Simply `delete`.  Fix only your touched code - fixing all pointless if/deletes below is out of scope of your PR.

Reply:
Done.